File: java/classes/constructors/ConstructorWithArguments.java
class Basket {
  public Basket(String basket_label, int basket_capacity) {
    label = basket_label;
    capacity = basket_capacity;
    items = new String[capacity];
    load_counter = 0;
  }

  public void put(String item) {
    if (load_counter < capacity) {
      items[load_counter] = item;
      load_counter++;
    }
  }

  public String pull() {
    String item;

    if (load_counter > 0) {
      item = items[load_counter - 1];
      items[load_counter - 1] = "";
      load_counter--;
    } else {
      item = "";
    }

    return item;
  }

  public void show() {
    System.out.print(label + ": ");
    for (int i = 0; i < load_counter; i++) {
      System.out.print(items[i] + " ");
    }
    System.out.println();
  }

  private String label;
  private final int capacity;
  private String[] items;
  private int load_counter;
}

class ConstructorWithArguments {
  public static void main(String args[]) {
    Basket handbasket = new Basket("fruits", 5);

    handbasket.put("apple");
    handbasket.put("pear");
    handbasket.put("plum");
    handbasket.put("peach");
    handbasket.put("grape");

    System.out.println("BASKET:");

    handbasket.show();

    System.out.println("REMOVED ITEMS:");

    System.out.println(handbasket.pull());
    System.out.println(handbasket.pull());
    System.out.println(handbasket.pull());

    System.out.println("BASKET:");

    handbasket.show();

    System.out.println("REMOVED ITEMS:");

    System.out.println(handbasket.pull());
    System.out.println(handbasket.pull());
    System.out.println(handbasket.pull());

    System.out.println("BASKET:");

    handbasket.show();
  }
}

File: javascript/classes/constructors/constructor_with_arguments.js
#!/usr/bin/env node

class Basket {
  constructor(label, capacity = 3) {
    this.label = label;
    this.capacity = capacity;
    this.load_counter = 0;
    this.items = [];
  }

  put = function(item) {
    if (this.load_counter < this.capacity) {
      this.items[this.load_counter] = item;
      this.load_counter++;
    }
  }

  pull = function() {
    let item;

    if (this.load_counter > 0) {
      item = this.items[this.load_counter - 1];
      this.items[this.load_counter - 1] = "";
      this.load_counter--;
    } else {
      item = "";
    }

    return item;
  }

  show = function() {
    process.stdout.write(this.label + ': ');
    for (let item of this.items) {
      process.stdout.write(item + ' ');
    }
    console.log();
  }

  label;
  capacity;
  items;
  load_counter;
}

let handbasket = new Basket("fruits", 5);

handbasket.put("apple");
handbasket.put("pear");
handbasket.put("plum");
handbasket.put("peach");
handbasket.put("grape");

console.log("BASKET:");

handbasket.show();

console.log("REMOVED ITEMS:");

console.log(handbasket.pull());
console.log(handbasket.pull());
console.log(handbasket.pull());

console.log("BASKET:");

handbasket.show();

console.log("REMOVED ITEMS:");

console.log(handbasket.pull());
console.log(handbasket.pull());
console.log(handbasket.pull());

console.log("BASKET:");

handbasket.show();
